### PR TITLE
replace some type comparisons in combinat/

### PR DESCRIPTION
**Pull request by fchapoton**
Replace some type comparison in 3 files in the combinat folder

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/combinat/growth.py [hunk 1]
```diff
@@ -1128,13 +1128,13 @@ def __eq__(self, other):
             sage: G1 == G2
             False
         """
-        return (type(self) == type(other) and
+        return (isinstance(other, GrowthDiagram) and
```
Comment (mkoeppe) [inline src/sage/combinat/growth.py:1131]: I don't think this is a correct replacement. For equality, the types really should be identical.

File: src/sage/combinat/words/paths.py [hunk 2]
```diff
@@ -455,12 +455,12 @@ def __eq__(self, other):
             sage: W1 == W3
             False
         """
-        return self is other or (type(self) == type(other) and
+        return self is other or (isinstance(other, WordPaths_all) and
```
Comment (mkoeppe) [inline src/sage/combinat/words/paths.py:458]: same comment as above.

**Comment by fchapoton** [comment]
well, may you please explain why you think so ? According to pycodestyle, one should avoid comparing types.

**Comment by mkoeppe** [comment]
Well, it's clearly intended in `__eq__` methods of this kind that objects of different type compare as different.
I don't think pycodestyle's rules can override the intended semantics.
To get rid of this warning, perhaps you can change it to `self.__class__ == other.__class__` but aside from pycodestyle, it's hard to see why that would be an improvement

**Pull request closed without merging**